dali/kernels/slice/slice_gpu.cuh

```diff
@@ -137,65 +132,39 @@ __device__ void SliceFuncNoPad(OutputType *__restrict__ out, const InputType *__
 /**
  * @brief General algorithm that allows for padding in any dimension
  * @remarks `in` refers to the beginning of the input (not the slice anchor)
- * @remarks `AllDims=true` means that Dims refer to the actual number of dimensions,
- *           meaning we haven't skipped last dimensions that have same input and output strides
  */
-template <int Dims, typename OutputType, typename InputType, bool AllDims = true>
+template <int Dims, typename OutputType, typename InputType>
 __device__ void SliceFunc(OutputType *__restrict__ out, const InputType *__restrict__ in,
                           const fast_div<uint64_t> *out_strides, const int64_t *in_strides,
                           const int64_t *out_shape, const int64_t *in_shape, const int64_t *anchor,
-                          const OutputType *__restrict__ fill_values, int channel_dim,
-                          uint64_t offset, uint64_t block_end) {
-  if (Dims > 1 && anchor[Dims - 1] == 0 && in_shape[Dims - 1] == out_shape[Dims - 1] &&
-      channel_dim != Dims - 1) {
```

Comment:
This could be kept, provided that we add a condition that `step[Dims - 1] == 0`.

Reply:
`step[Dims - 2]` was the missing piece I didn't get, I tried with `step[Dims -1] == 1` and still had malformed outputs which I why I removed flattening. Adding `step[Dims - 2] == 1` fixes this (I've tested test_subscipt so far and all passes, running the remainder of operators_2 now.).

Reply:
Yeah, the anchor and extent are adjusted, but you cannot adjust the step the same way.
Imagine that you have an RGB image and want to select `[top:bottom][left:right:2][:]`
You want to get every second RGB block - this cannot be achieved when the loop over channels is fused with the loop over pixels.
You want:
```
RGB rgb RGB rgb RGB rgb
^^^     ^^^     ^^^
```
With fusion you get:
```
RGB rgb RGB rgb RGB rgb
^ ^  ^  ^ ^  ^  ^ ^  ^ 
```
The step, being a single number, cannot be adjusted the way anchor and extent can.

Reply:
Since most of the tests are only 2D arrays, I just checked 3,4,5 and that seems to be fine as well. I might think of making a super-test that does a bunch of permutations of data dimensions, start, end, step, slice dimension. Right now its hard to do without a lot of code duplication.

Although you end up with the problem of "spend 1hr to design n-d testing" vs "spend 10min to copy-paste make 1 change for each test".

Reply:
I advocate simpler tests (even if redundant) for clarity. A test should not involve more complex logic than the program being tested.
Regarding the `step[Dims - 2] == 1` - I still don't see it in the last code drop.